### PR TITLE
Google Lighthouse Fixes

### PR DESCRIPTION
**Pull request by Charlesthebird**
# Description
This PR includes fixes related to Google Lighthouse testing on the home and APIs pages. We have less control over the API details page since we use the Redoc and Swagger libraries for viewing the API schema, but these improvements carried over to that page as well. 
- The default images we include were converted from PNG to webp. This ended up being a big improvement.
- Image aspect ratios are now maintained using the [object-fit](https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit) CSS property.
- Invalid `role` attributes were removed.

# Results

Some of the suggestions are related to what the `VITE_PORTAL_SERVER_URL` and image environment variables are set to, so those scores will depend on user configurations.

The home page and APIs page had very similar scores, and improvements, so I will just include the home page scores here.

Before lighthouse fixes (on the home page):
[lighthouse-before-improvements.pdf](https://github.com/user-attachments/files/16315307/lighthouse-before-improvements.pdf)
- Performance: 83
- Accessibility: 88
- Best Practices: 70
- The build size is 5MB

After lighthouse fixes (on the home page):
[lighthouse-after-improvements.pdf](https://github.com/user-attachments/files/16315331/lighthouse-after-improvements.pdf)
- Performance: 91
- Accessibility: 95
- Best Practices: 78
- The build size is 4.2MB

BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh-enterprise/issues/15009

### CODE REVIEW

File: projects/ui/index.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html>
+<!doctype html>
```
Comment (toreysoloio) [inline projects/ui/index.html:1]: huh; I never knew this should be lowercase

File: projects/ui/src/Components/Home/HomePage.style.tsx [hunk 2]
```diff
@@ -87,10 +87,9 @@ export namespace HomePageStyles {
       // This is a best-guess background fit attempt for user-supplied images.
       height: 100%;
       ${mediaQueryWithScreenSize.mediumAndSmaller} {
-        height: unset;
         width: 100%;
       }
-      background-size: cover;
+      object-fit: cover;
```
Comment (toreysoloio) [inline projects/ui/src/Components/Home/HomePage.style.tsx:92]: Very nice! I actually used this once on a personal project once ages ago, but I 1000% forgot this existed; thanks for the reminder ;)

**Review by toreysoloio — APPROVED** [review]

File: projects/ui/index.html (hunk 1 above)
Comment (Charlesthebird) [reply projects/ui/index.html:1]: I think this was actually a prettier fix, and not in the lighthouse score.

Comment (Charlesthebird) [reply projects/ui/index.html:1]: Apparently it's not case sensitive, so it should be okay either way.

**Pull request merged**